### PR TITLE
Rework frontend errors

### PR DESCRIPTION
**Pull request by W95Psp**
This PR provides a more uniform way of reporting errors in the exporter.

### CODE REVIEW

File: frontend/exporter/src/utils.rs [hunk 1]
```diff
@@ -1,57 +1,107 @@
...
+            $verb,
+            $s,
+            $s.base().opt_def_id.map(|did| $s.base().tcx.def_span(did)),
+            $crate::utils::format_with_context!($($tt)*)
+        )
+    };
+}
 
-Please report that message along with the crate!
+#[allow(unused_macros)]
```
Comment (franziskuskiefer) [inline frontend/exporter/src/utils.rs:60]: Are these all unused or why do we need the `allow`s here?

File: frontend/exporter/src/utils.rs [hunk 2]
```diff
@@ -1,57 +1,107 @@
...
-        }}
+    macro_rules! _span_verb_base {
+        ($verb:ident, $s:ident, $span:expr, $message:expr) => {{
+            let mut builder = $crate::utils::_verb!($verb, $s.base().tcx.sess, $message);
+            if let Some(span) = $span {
+                builder.set_span(span.clone());
+            }
+            builder.code(rustc_errors::DiagnosticId::Error(format!("HaxFront")));
+            builder.note(
+                "⚠️ This is a bug in Hax's frontend.
```
Comment (franziskuskiefer) [inline frontend/exporter/src/utils.rs:35]: Add a linkt to the issues in this crate.

**Review by franziskuskiefer — APPROVED** [review]
Thanks

File: frontend/exporter/src/utils.rs (hunk 1 above)
Comment (W95Psp) [reply frontend/exporter/src/utils.rs]: Only `warning` is unused, actually, removing those other superfluous `allow`s!

**Pull request merged**